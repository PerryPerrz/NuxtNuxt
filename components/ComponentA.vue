<template>
    <div>
        <h2>Component A</h2>
        <p>Composant dynamik et KeepAlive. (A)</p>

        <div class="content">
            <p>Counter : {{ cpt }}</p>
            <button style="margin-right: 1rem;" @click="increment" data-type="increment">Increment</button>
            <button @click="decrement" data-type="decrement">Decrement</button>
        </div>
    </div>
</template>

<script setup lang="ts">
    import { ref, onMounted } from 'vue';
    
    onMounted(() => {
        console.log('ComponentA mounted');
    });

    const cpt = ref<number>(0);
    const increment = () => { cpt.value++;};
    const decrement = () => { cpt.value--;};
</script>

<style scoped>
.content {
    margin-top: 1rem;
    margin-bottom: 1rem;
    padding: 0.5rem;
    border: 1px solid #ccc;
    border-radius: 0.5rem;
    background-color: #f9f9f9;
}
</style>