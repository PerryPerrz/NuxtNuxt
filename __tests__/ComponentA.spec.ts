import ComponentA from '../components/ComponentA.vue';
import { describe, test, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

describe('ComponentA', () => {
    test('ComponentA is defined', () => {
        expect(ComponentA).toBeDefined();
    });
});

describe('ComponentA functions', () => {
    const wrapper = mount(ComponentA);

    it('should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render correctly bis', () => {
        expect(wrapper.html()).toContain('Component A');
        expect(wrapper.html()).toContain('KeepAlive');
        expect(wrapper.html()).toContain('Counter : 0');
    })

    it('should increment counter', async () => {
        /*
        const button = wrapper.find('button');
        await button.trigger('click');
        expect(wrapper.html()).toContain('Counter : 1');*/

        const btnIncrement = wrapper.find('[data-type="increment"]');
        await btnIncrement.trigger('click');
        expect(wrapper.html()).toContain('Counter : 1');
    });

    it('should decrement counter', async () => {
        const btnDecrement = wrapper.find('[data-type="decrement"]');
        await btnDecrement.trigger('click');
        expect(wrapper.html()).toContain('Counter : 0');
    });

    it('should decrement counter bis', async () => {
        const btnDecrement = wrapper.find('[data-type="decrement"]');
        await btnDecrement.trigger('click');
        expect(wrapper.html()).toContain('Counter : -1');
    });

    it('should increment counter bis', async () => {
        const btnIncrement = wrapper.find('[data-type="increment"]');
        await btnIncrement.trigger('click');
        expect(wrapper.html()).toContain('Counter : 0');
    });
});