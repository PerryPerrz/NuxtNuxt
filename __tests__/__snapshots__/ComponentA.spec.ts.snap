// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ComponentA functions > should render correctly 1`] = `
"<div data-v-38e6153d="">
  <h2 data-v-38e6153d="">Component A</h2>
  <p data-v-38e6153d="">Composant dynamik et KeepAlive. (A)</p>
  <div data-v-38e6153d="" class="content">
    <p data-v-38e6153d="">Counter : 0</p><button data-v-38e6153d="" style="margin-right: 1rem;" data-type="increment">Increment</button><button data-v-38e6153d="" data-type="decrement">Decrement</button>
  </div>
</div>"
`;
